spec/unit/functions/lookup_spec.rb

```diff
@@ -0,0 +1,263 @@
+#! /usr/bin/env ruby
+require 'spec_helper'
+require 'puppet_spec/compiler'
+require 'puppet/pops'
+
+describe "when performing lookup" do
+  include PuppetSpec::Compiler
+
+  # Assembles code that includes the *abc* class and compiles it into a catalog. This class will use the global
+  # variable $args to perform a lookup and assign the result to $abc::result. The assembled code will declare
+  # a notify resource with a name that is formed by interpolating the result into a format string.
+  #
+  # The method performs the folloging steps.
+  #
+  # - Build the code that:
+  #    - sets the $args variable from _lookup_args_
+  #    - includes the abc class
+  #    - assigns the $abc::result to $r
+  #    - interpolates a string using _fmt_ (which is assumed to use $r)
+  #    - declares a notify resource from the interpolated string
+  # - Compile the code into a catalog
+  # - Return the name of all Notify resources in that catalog
+  #
+  # @param fmt [String] The puppet interpolated string used when creating the notify name
+  # @param *args [String] splat of args that will be concatenated to form the puppet args sent to lookup
+  # @return [Array<String>] List of names of Notify resources in the resulting catalog
+  #
+  def assemble_and_compile(fmt, *lookup_args)
+    Puppet[:code] = \
```

Comment:
you can use a heredoc with interpolation here instead to make it easier to read than the string continuations